x/ibc/core/24-host/parse.go

```diff
@@ -1,11 +1,35 @@
 package host
 
 import (
+	"strconv"
 	"strings"
 
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
 )
 
+// ParseIdentifier parses the sequence from the identifier using the provided prefix.
+func ParseIdentifier(identifier, prefix string) (uint64, error) {
```

Comment:
Document that these identifier formats are only expected on our own chain. Counterparty identifiers must simply pass the old validation functions